custom_components/enphase_envoy_custom/__init__.py

```diff
@@ -74,9 +74,11 @@ async def async_update_data():
                         battery_sum = 0
                         for battery in battery_data:
                             battery_sum += battery.get("percentFull", 0)
-
-                        data[description.key] = round(battery_sum / len(battery_data), 2)
-
+                        try:
```

Comment:
IMO a better way to fix this is to add `and len(battery_data) > 0` to the if statements on line 73 and 84. This will prevent the divide by zero and also prevent the current battery capacity entity from being added when there are no batteries.

Reply:
In trying your len(battery_data) strategy, another error gets generated further down about the current_battery_capacity not being defined, IIRC.

Reply:
Can you paste the error you get regarding `current_battery_capacity`?

Reply:
Sorry I haven't had a chance to look into this, been very busy on my side. I made multiple changes to get around the error, just to move past it.